### PR TITLE
Fix export and test buttons

### PR DESCRIPTION
**Pull request by Jean28518**
Should now work as expected.
Now testing and exporting is available from the scenario editor too.
Created a new export_track class to keep to avoid code duplications.

Does NOT implement #187

### CODE REVIEW

File: src/Editor/Scripts/Editor.gd [hunk 1]
```diff
@@ -251,7 +251,6 @@ func clear_selected_object() -> void:
 			vi.set_layer_mask_bit(1, false)
 
 		if selected_object_type == "Building":
-			selected_object.get_child(1).queue_free()
```
Comment (HaSa1002) [inline src/Editor/Scripts/Editor.gd:254]: Why is that? Can you still delete a building?

File: src/Editor/Scripts/Editor.gd [hunk 2]
```diff
@@ -454,6 +453,11 @@ func test_track_pck() -> void:
 		send_message("Can't test tracks if runs Libre TrainSim using the Godot Editor. " \
 				+ "Please use a build of Libre TrainSim to test tracks. ")
 		return
+
+	if ContentLoader.get_scenarios_for_track(current_track_path).size() == 0:
+		send_message("To test this track you need to create a scenario in the scenario editor first")
```
Comment (HaSa1002) [inline src/Editor/Scripts/Editor.gd]:
```suggestion
		send_message("Cannot test the track! Please create a scenario using the scenario editor.")
```

File: src/Editor/Scripts/Editor.gd [hunk 3]
```diff
@@ -465,62 +469,18 @@ func test_track_pck() -> void:
...
-	while file_name != "":
-		if dir.current_is_dir():
-			files.append_array(get_files_in_directory(path.plus_file(file_name)))
-		else:
-			files.append(path.plus_file(file_name))
-		file_name = dir.get_next()
-	return files
+func export_mod() -> String:
+	if ContentLoader.get_scenarios_for_track(current_track_path).size() == 0:
+		return("No scenario found! To export this track you need to create a scenario in the scenario editor first")
```
Comment (HaSa1002) [inline src/Editor/Scripts/Editor.gd]:
```suggestion
		send_message("No scenario found! Please create a scenario in the scenario editor to enable track exporting.")
		return
```
Please send the message in the function and return void instead.

File: src/Editor/Scripts/Editor.gd [hunk 4]
```diff
@@ -465,62 +469,18 @@ func test_track_pck() -> void:
...
 
+	save_world(false)
+	var mod_name = current_track_path.get_file().get_basename()
+	var export_path = "user://addons/"
+	return ExportTrack.export_editor_track(mod_name, export_path)
 
 
 func _on_ExportTrack_pressed() -> void:
-	export_mod()
+	send_message(export_mod())
```
Comment (HaSa1002) [inline src/Editor/Scripts/Editor.gd]:
```suggestion
	export_mod()
```

File: src/Editor/Scripts/export_track.gd [hunk 5]
```diff
@@ -0,0 +1,39 @@
+class_name ExportTrack
```
Comment (HaSa1002) [inline src/Editor/Scripts/export_track.gd:1]: I assume this is the code that was in editor.gd? Please compare and check with the mod_tools script. In doubt the exporting of the mod tools is correct and should be used instead.

File: src/Editor/Scripts/scenario_editor.gd [hunk 6]
```diff
@@ -74,3 +83,47 @@ func run_map_updater(delta: float):
...
 		$ScenarioMap.update_map()
+
+
+func _on_TestTrack_pressed():
+	$CanvasLayer/Pause.hide()
+	test_track_pck()
+
+func _on_ExportTrack_pressed():
+	$CanvasLayer/Pause.hide()
+	show_message(export_mod())
```
Comment (HaSa1002) [inline src/Editor/Scripts/scenario_editor.gd]:
```suggestion
	export_mod()
```

File: src/Editor/Scripts/scenario_editor.gd [hunk 7]
```diff
@@ -74,3 +83,47 @@ func run_map_updater(delta: float):
...
+
+
+func test_track_pck() -> void:
+	if OS.has_feature("editor"):
+		show_message("Can't test tracks if runs Libre TrainSim using the Godot Editor. " \
+				+ "Please use a build of Libre TrainSim to test tracks. ")
+		return
+
+	if ContentLoader.get_scenarios_for_track(Root.current_editor_track_path).size() == 0:
+		show_message("To test this track you need to create a scenario in the scenario editor first")
```
Comment (HaSa1002) [inline src/Editor/Scripts/scenario_editor.gd]:
```suggestion
		send_message("Cannot test the track! Please create a scenario.")
```

File: src/Editor/Scripts/scenario_editor.gd [hunk 8]
```diff
@@ -74,3 +83,47 @@ func run_map_updater(delta: float):
...
+		show_message("Can't load content pack!")
+		return
+	ContentLoader.append_content_to_global_repo(content)
+	Root.start_menu_in_play_menu = true
+	LoadingScreen.load_main_menu()
+
+
+func export_mod() -> String:
+	if ContentLoader.get_scenarios_for_track(current_track_path).size() == 0:
+		return "No scenario found! To export this track you need to create a scenario in the scenario editor first"
```
Comment (HaSa1002) [inline src/Editor/Scripts/scenario_editor.gd]:
```suggestion
func export_mod() -> void:
	if ContentLoader.get_scenarios_for_track(current_track_path).size() == 0:
		send_message("No scenario found! Please create a scenario.")
		return
```

File: src/Editor/Scripts/scenario_editor.gd [hunk 9]
```diff
@@ -74,3 +83,47 @@ func run_map_updater(delta: float):
...
+	save_scenario()
+	var track_name = current_track_path.get_file().get_basename()
+	var export_path = "user://addons/"
+	return ExportTrack.export_editor_track(track_name, export_path)
```
Comment (HaSa1002) [inline src/Editor/Scripts/scenario_editor.gd]:
```suggestion
	send_message(ExportTrack.export_editor_track(track_name, export_path))
	return
```

**Review by HaSa1002 — CHANGES_REQUESTED** [review]

File: src/Editor/Scripts/export_track.gd (hunk 5 above)
Comment (Jean28518) [reply src/Editor/Scripts/export_track.gd:1]: Tried this already.
But didn't work out.
I think it was about different pathes somewhere.

File: src/Editor/Scripts/scenario_editor.gd (hunk 6 above)
Comment (Jean28518) [reply src/Editor/Scripts/scenario_editor.gd:94]: Why?

File: src/Editor/Scripts/Editor.gd (hunk 4 above)
Comment (Jean28518) [reply src/Editor/Scripts/Editor.gd:484]: Why?

File: src/Editor/Scripts/Editor.gd (hunk 1 above)
Comment (Jean28518) [reply src/Editor/Scripts/Editor.gd:254]: Thanks for your hint.
This change was wrong of me. It was crashing there.

Reimplemented it without any crashes in the latest commit.

**Comment by Jean28518** [comment]
For me this one can be merged.

**Review by HaSa1002 — APPROVED** [review]
untested. Let's merge and see what breaks :P

**Pull request merged**